docs: fix minor typos and add links

lumirlumir opened the pull request:
<!--
    Thank you for contributing!

    ESLint adheres to the [OpenJS Foundation Code of Conduct](https://eslint.org/conduct).
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

<!--
    The following template is intentionally not a markdown checkbox list for the reasons
    explained in https://github.com/eslint/eslint/pull/12848#issuecomment-580302888
-->

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

<!--
    If the item you've checked above has a template, please paste the template questions below and answer them. (If this pull request is addressing an issue, you can just paste a link to the issue here instead.)
-->

<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/latest/contribute/pull-requests)
    - Include tests for this change
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

#### What changes did you make? (Give an overview)

Hello,

In this PR, I’ve fixed a few minor typos and added relevant links to improve readability.

#### Is there anything you'd like reviewers to focus on?

<!-- markdownlint-disable-file MD004 -->

lumirlumir commented on docs/src/maintain/manage-issues.md:
```diff
@@ -44,7 +44,7 @@ All of ESLint's issues and pull requests, across all GitHub repositories, are ma
 - **Implementing**: There is an open pull request for each of these issues or this is a pull request that has been approved.
 - **Second Review Needed**: Pull requests that already have one approval and the approver is requesting a second review before merging.
 - **Merge Candidates**: Pull requests that already have at least one approval and at least one approver believes the pull request is ready to merge into the next release but would still like a TSC member to verify.
-- **Completed**: The issue has been closed (either via pull request merge or by the team manually closing the issue).
```
![image](https://github.com/user-attachments/assets/303dbb3f-a72f-4438-9115-b92b4d4d51b6)

lumirlumir commented on docs/src/use/core-concepts/index.md:
```diff
@@ -52,11 +52,11 @@ Shareable configurations are ESLint configurations that are shared via npm.
 
 Often shareable configurations are used to enforce style guides using ESLint's built-in rules. For example the sharable configuration [eslint-config-airbnb-base](https://www.npmjs.com/package/eslint-config-airbnb-base) implements the popular Airbnb JavaScript style guide.
 
-For more information, refer to [Using a shareable configuration package](../configure/configuration-files#using-a-shareable-configuration-package).
```
![image](https://github.com/user-attachments/assets/321ce766-e1a8-4de7-bf3c-52158a4e059c)

Other links use capital letters for their first words, so this one should be consistent as well.

mdjermanovic commented on docs/src/use/core-concepts/glossary.md:
```diff
@@ -187,7 +187,7 @@ You can let ESLint know which global variables your code uses with [global decla
 
 Inline configs use similar syntax as [config files](#config-file-configuration-file) to specify any number of rules by name, their new severity, and optionally new options for the rules.
-For example, the following inline config comment simultaneously disables the `eqeqeq` rule and sets the `curly` rule to `"error"`:
+For example, the following inline config comment simultaneously disables the [`eqeqeq`](https://eslint.org/docs/latest/rules/eqeqeq) rule and sets the [`curly`](https://eslint.org/docs/latest/rules/curly) rule to `"error"`:
```
We should use relative URLs that start with `../`, because we have multiple docs sites, for different versions of ESLint, so a docs page for one version should point to a docs page for the same version.

mdjermanovic commented on docs/src/use/core-concepts/index.md:
```diff
@@ -52,11 +52,11 @@ Shareable configurations are ESLint configurations that are shared via npm.
 ## Plugins
 
-An ESLint plugin is an npm module that can contain a set of ESLint rules, configurations, processors, and languages. Often plugins include custom rules. Plugins can be used to enforce a style guide and support JavaScript extensions (like TypeScript), libraries (like React), and frameworks (Angular).
+An ESLint plugin is an npm module that can contain a set of ESLint rules, configurations, processors, and languages. Often plugins include custom rules. Plugins can be used to enforce a style guide and support JavaScript extensions (like [TypeScript](https://www.typescriptlang.org)), libraries (like [React](https://react.dev)), and frameworks ([Angular](https://angular.dev)).
```
```suggestion
An ESLint plugin is an npm module that can contain a set of ESLint rules, configurations, processors, and languages. Often plugins include custom rules. Plugins can be used to enforce a style guide and support JavaScript extensions (like [TypeScript](https://www.typescriptlang.org)), libraries (like [React](https://react.dev)), and frameworks (like [Angular](https://angular.dev)).
```

While we are here, seems like "like" was missing here.

lumirlumir commented:
@snitin315 @mdjermanovic

Thank you for the guidance!

I’ve addressed all the comments, and I also found a few more typos and areas for improvement, I’ve added additional commits  https://github.com/eslint/eslint/pull/19743/commits/a4f8866dfca534cd993b870c155b5c30b038f098.

mdjermanovic reviewed (approved):
LGTM, thanks! Leaving open for @snitin315 to verify.

snitin315 reviewed (approved):
LGTM, thanks!
